### ISSUE TITLE
`consistent-boolean-name` should allow Promise<boolean>

### ISSUE THREAD

**Issue by Sparticuz** [sindresorhus/eslint-plugin-unicorn#3500]
I thought it was working in v69, but after updating to v71 it's now erroring. This function returns a boolean, so it shouldn't trip the rule

`consistent-boolean-name`

```ts
export const isInsideSale = async (salesRep: string | null) => {
  if (salesRep) {
    const [user] = await database
      .select()
      .from(employees)
      .where(eq(employees.userName, salesRep));
    if (!user) {
      return false;
    }
  }
  return true;
};
```
